File: hr-connect-app/app/src/main/java/com/hrconnect/app/BootReceiver.java
package com.hrconnect.app;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import org.json.JSONObject;

/** Restarts the WiFi auto-clock service after the phone reboots. */
public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (!Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) return;
        JSONObject cfg = WifiClockService.readConfig(context);
        if (cfg != null && cfg.optBoolean("enabled", false)) {
            WifiClockService.start(context);
        }
    }
}

File: Perimeter/app/src/main/java/com/perimeter/attendance/service/BootReceiver.kt
package com.perimeter.attendance.service

import android.Manifest
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.pm.PackageManager
import androidx.core.content.ContextCompat

/**
 * The service has to come back after a reboot or an app update, or a staff
 * member who restarts their phone silently stops being clocked in.
 */
class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent?) {
        val granted = ContextCompat.checkSelfPermission(
            context, Manifest.permission.ACCESS_FINE_LOCATION
        ) == PackageManager.PERMISSION_GRANTED
        if (granted) AttendanceService.start(context)
    }
}
